src/object_store/src/object/opendal_engine/opendal_object_store.rs

```diff
@@ -38,6 +38,7 @@ pub enum EngineType {
     Memory,
     Hdfs,
     Gcs,
+    OpendalS3,
```

Comment:
nits: It is better to name it `S3` so that it is consistent with other EngineType